Formula/swi-prolog.rb

```diff
@@ -69,7 +69,7 @@ def install
     # './prepare' prompts the user to build documentation
     # (which requires other modules). '3' is the option
     # to ignore documentation.
-    system "echo '3' | ./prepare" if build.head?
+    system "echo", "'3' | ./prepare" if build.head?
```

Comment:
> It looks like system calls with `|` in them may supposed to be exempt [already](https://github.com/Homebrew/brew/blob/a0d008af77785071ffba27dd995e5c794042f649/Library/Homebrew/cmd/audit.rb#L908).

@DomT4 That check trips over the quotes in quotes issue, because it doesn't use a back-reference to the opening quote character to find the end and thus stops capturing before it hits the pipe (`|`):

```
swi-prolog:
  * Use `system "echo", "'` instead of `system "echo '`
```

Of course the inner single quotes are completely pointless in this case and can be omitted, which will also silence the false warning without altering behavior. (Addressed in 7039eedc2fd645f2dfa90925f745b83fb29f2199.)